src/Bridges/Constraint/bridges/integer_to_zeroone.jl

```diff
@@ -0,0 +1,168 @@
+            MOI.ConstraintIndex{MOI.ScalarAffineFunction{T},MOI.EqualTo{T}}(0),
+            nothing,
+        )
+    end
+end
+
+const IntegerToZeroOne{T,OT<:MOI.ModelLike} =
+    SingleBridgeOptimizer{IntegerToZeroOneBridge{T},OT}
+
+function bridge_constraint(
+    ::Type{IntegerToZeroOneBridge{T}},
+    ::MOI.ModelLike,
+    x::MOI.VariableIndex,
+    ::MOI.Integer,
+) where {T}
+    # !!! info
+    #     Postpone creation until final_touch.
+    return IntegerToZeroOneBridge{T}(x)
+end
+
+function MOI.supports_constraint(
+    ::Type{IntegerToZeroOneBridge{T}},
+    ::Type{MOI.VariableIndex},
+    ::Type{MOI.Integer},
+) where {T}
+    return true
+end
+
+function MOI.Bridges.added_constrained_variable_types(
+    ::Type{<:IntegerToZeroOneBridge},
+)
+    return Tuple{Type}[(MOI.ZeroOne,)]
+end
+
+function MOI.Bridges.added_constraint_types(
+    ::Type{IntegerToZeroOneBridge{T}},
+) where {T}
+    return Tuple{Type,Type}[(MOI.ScalarAffineFunction{T}, MOI.EqualTo{T})]
+end
+
+function concrete_bridge_type(
+    ::Type{IntegerToZeroOneBridge{T}},
+    ::Type{MOI.VariableIndex},
+    ::Type{MOI.Integer},
+) where {T}
+    return IntegerToZeroOneBridge{T}
+end
+
+function MOI.get(
+    ::MOI.ModelLike,
+    ::MOI.ConstraintFunction,
+    bridge::IntegerToZeroOneBridge,
+)
+    return bridge.x
+end
+
+function MOI.get(::MOI.ModelLike, ::MOI.ConstraintSet, ::IntegerToZeroOneBridge)
+    return MOI.Integer()
+end
+
+function MOI.delete(model::MOI.ModelLike, bridge::IntegerToZeroOneBridge)
+    MOI.delete(model, bridge.ci)
+    MOI.delete(model, bridge.y)
+    return
+end
+
+function MOI.get(bridge::IntegerToZeroOneBridge, ::MOI.NumberOfVariables)::Int64
+    return length(bridge.y)
+end
+
+function MOI.get(bridge::IntegerToZeroOneBridge, ::MOI.ListOfVariableIndices)
+    return copy(bridge.y)
+end
+
+function MOI.get(
+    bridge::IntegerToZeroOneBridge,
+    ::MOI.NumberOfConstraints{MOI.VariableIndex,MOI.ZeroOne},
+)::Int64
+    return length(bridge.y)
+end
+
+function MOI.get(
+    bridge::IntegerToZeroOneBridge,
+    ::MOI.ListOfConstraintIndices{MOI.VariableIndex,MOI.ZeroOne},
+)
+    return map(bridge.y) do y
+        return MOI.ConstraintIndex{MOI.VariableIndex,MOI.ZeroOne}(y.value)
+    end
+end
+
+function MOI.get(
+    bridge::IntegerToZeroOneBridge{T},
+    ::MOI.NumberOfConstraints{MOI.ScalarAffineFunction{T},MOI.EqualTo{T}},
+)::Int64 where {T}
+    return 1
+end
+
+function MOI.get(
+    bridge::IntegerToZeroOneBridge{T},
+    ::MOI.ListOfConstraintIndices{MOI.ScalarAffineFunction{T},MOI.EqualTo{T}},
+) where {T}
+    return [bridge.ci]
+end
+
+MOI.Bridges.needs_final_touch(::IntegerToZeroOneBridge) = true
+
+function MOI.Bridges.final_touch(
+    bridge::IntegerToZeroOneBridge{T},
+    model::MOI.ModelLike,
+) where {T}
+    ret = MOI.Utilities.get_bounds(model, T, bridge.x)
+    if ret === bridge.last_bounds
+        return nothing  # final_touch already called
+    elseif ret[1] == typemin(T) || ret[2] == typemax(T)
```

Comment:
We should probably have a utility for that shared with `SplitInterval` and other bridges doing this like `is_infinity`. In `Interval` we take care of `Int` while we don't do it here.

Reply:
I don't understand the SplitInterval reference. The SplitInterval bridge doesn't rely on a finite domain, it works with one-sided intervals. In comparison, we can't bridge an open set to ZeroOne (unless we pick a big M and do a 20 or 30 bit expansion?).